app/controllers/events/view/export.js

```diff
@@ -0,0 +1,48 @@
+import Ember from 'ember';
+
+const { Controller, run, $ } = Ember;
+export default Controller.extend({
+  eventExportStatus : '',
+  eventDownloadUrl  : '',
+  actions           : {
+    startGeneration() {
+      let payload = {
+        image    : this.get('dataImage'),
+        video    : this.get('dataVideo'),
+        audio    : this.get('dataAudio'),
+        document : this.get('dataDocument')
+      };
+      let exportUrl = `/events/${this.get('model').id}/export/json`;
+      this.get('loader')
+        .post(exportUrl, payload)
+        .then(data => {
+          let requestLoop = run.later(() => {
+            this.get('loader')
+              .load(data.task_url, { withoutPrefix: true })
+              .then(zipDownloadUrl => {
+                if (zipDownloadUrl.state === 'SUCCESS') {
+                  $('#eventDownloadButton').removeClass('disabled');
+                  this.set('eventExportStatus', 'SUCCESS');
```

Comment:
If you're anyway setting the state, you could just do a `this.set('eventExportStatus', zipDownloadUrl.state)` in a common place right ?